roles/zuul/tasks/main.yml

```diff
@@ -136,6 +136,17 @@
   include: merger.yml
   when: "'zuul-merger' in zuul_components"
 
+- name: Stop unwanted zuul services
+  service:
+    name: "{{ item }}"
+    state: stopped
+    enabled: no
+  when: "item not in zuul_components"
+  with_items:
+    - zuul-launcher
+    - zuul-merger
+    - zuul-server
```

Comment:
I was thinking more what about in the like all-in-one case where zuul_components may be specified on a role or something and executed multiple times. does that still work?